Provide char16_t and char32_t definitions for ONIE

Copyright (C) 2020 Alex Doyle <adoyle@nvidia.com>
Copyright (C) 2015 Curt Brune <curt@cumulusnetworks.com>

SPDX-License-Identifier:     GPL-2.0

Since uClibc does not provide <uchar.h> provide explicit typedefs for
char16_t and char32_t using the compiler's internal definitions.
Updated patch for efivar 37 - Alex
Summary:

diff --git a/src/export.c b/src/export.c
index 8321bd3..3ac9316 100644
--- a/src/export.c
+++ b/src/export.c
@@ -23,6 +23,19 @@
 #include <inttypes.h>
 #include <stdint.h>
 #include <stdlib.h>
+
+#ifdef __UCLIBC__
+/*
+ * Define the 16-bit and 32-bit character types as <uchar.h> is
+ * missing from uClibc.	 Use the information provided by the compiler
+ */
+#if !defined __CHAR16_TYPE__ || !defined __CHAR32_TYPE__
+#  error "definitions of __CHAR16_TYPE__ and/or __CHAR32_TYPE__ missing"
+#endif
+typedef __CHAR16_TYPE__ char16_t;
+typedef __CHAR32_TYPE__ char32_t;
+#else
 #include <uchar.h>
+#endif
 
 #include "efivar.h"
